extensions/README.md

```diff
@@ -0,0 +1,94 @@
+---
+tags: oci,distribution
+breaks: false
+---
+
+# Extensions API for Distribution
+
+Extensions, in general, to the distribution-spec API are allowed by the distribution-spec, with certain reservations disclosed herein.
+This document outlines a path for registry implementations and clients to iterate on new APIs, establish them, and propose them to OCI for canonical reference.
+
+The basis of the Extension API (`_oci`) should be emulated for all extensions.
+
+## Table
+
+_notice_: All new `/extensions/_$extention.md` docs MUST be added to this table.
+
+| Extension                | Summary                                              |
+|:------------------------:|:----------------------------------------------------:|
+| [`_oci`](./_oci.md)      | Extensions discovering extensions on registry server |
+| `_catalog`               | Reserved prior extension                             |
+
+## Name
+
+Extension names MUST be unique.
+Extensions recorded in this distribution-spec are considered canonical definitions.
+
+Extensions are specified by extension name (`$extension`) aligning with the project, followed by the `component`, and last by by the `module`.
+This constitutes the URI segments of the extension endpoint.
+Additional options may be passed as parameters to the endpoint.
+
+```http
+_<extension>/<component>/<module>[?<key>=<value>&...]
+```
+
+### Registry-Level Extensions
+
+Registry-level extensions are nested under `/v2`.
+
+```http
+GET /v2/_<extension>/<component>/<module>[?<key>=<value>&...]
+```
+
+A contrived example of a search extension may be of the form `/v2/_oci/catalog/search?pattern=foo?n=10`
+
+### Repository-Level Extensions
+
+Repository-level extension endpoints are scoped under a repository name.
+
+Repository-level extensions follow the same naming rules as the [registry-level endpoints](#registry-level-extensions).
+
+```http
+GET /v2/<name>/_<extension>/<component>/<module>[?<key>=<value>&...]
+```
+
+### Reserved Namespaces
+
+As of current, `_oci` and `_catalog` are considered as reserved namespaces and cannot be used by other extensions.
```

Comment:
... or as part of repository names.

Reply:
oh for sure. In the updated push I include some string formatting regular expression. With the `_` prefix on the extension name, that differentiates all extensions, from any `<name>` of a manifest reference.